geolocator_android/android/src/main/java/com/baseflow/geolocator/MethodCallHandlerImpl.java

```
@@ -238,7 +257,24 @@ private void onGetCurrentPosition(MethodCall call, MethodChannel.Result result)
 
           replySubmitted[0] = true;
           geolocationManager.stopPositionUpdates(locationClient);
+          pendingCurrentPositionLocationClients.remove(requestId);
           result.error(errorCode.toString(), errorCode.toDescription(), null);
         });
   }
+
+    /**
+     * Cancels a request for the current position that was initiated with {@link #onGetCurrentPosition}.
+     */
+    private void onCancelGetCurrentPosition(MethodCall call, MethodChannel.Result result) {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> arguments = (Map<String, Object>) call.arguments;
+      String requestId = (String) arguments.get("requestId");
+
+      LocationClient locationClient = this.pendingCurrentPositionLocationClients.get(requestId);
+      if (locationClient != null) {
+        locationClient.stopPositionUpdates();
+      }
+
+      result.success(null);
```

Fail silently or call `result.error()`?